### PR TITLE
Jb48360

### PR DESCRIPTION
**Pull request by jlehtoranta**
Only the last 3 commits belong to this PR

### CODE REVIEW

File: src/plugins/mer/mersdkmanager.cpp [hunk 1]
```diff
@@ -135,11 +139,13 @@ void MerSdkManager::initialize()
...
 
-        // If debugger became available
+        // If debugger and/or cmake became available
         for (BuildEngine *const engine : Sdk::buildEngines()) {
             for (const QString &targetName : engine->buildTargetNames()) {
-                for (Kit *const kit : kitsForTarget(targetName))
+                for (Kit *const kit : kitsForTarget(targetName)) {
                     ensureDebuggerIsSet(kit, engine, targetName);
+                    ensureCmakeToolIsSet(kit, engine, targetName);
+                }
```
Comment (martyone) [inline src/plugins/mer/mersdkmanager.cpp]: I wonder what makes cmake more similar to debugger than qmake w.r.t the way it is handled here?

File: src/plugins/mer/mersdkmanager.cpp [hunk 2]
```diff
@@ -43,6 +43,9 @@
 #include <coreplugin/coreconstants.h>
 #include <coreplugin/icore.h>
 #include <coreplugin/messagemanager.h>
+#include <cmakeprojectmanager/cmakekitinformation.h>
+#include <cmakeprojectmanager/cmaketool.h>
+#include <cmakeprojectmanager/cmaketoolmanager.h>
```
Comment (vigejolla) [inline src/plugins/mer/mersdkmanager.cpp]: Nitpick: Shouldn't these be before coreplugin includes?

File: src/tools/merssh/command.cpp [hunk 3]
```diff
@@ -44,6 +44,18 @@ int Command::executeRemoteCommand(const QString &command)
     // Convert to build engine paths
     QString mappedCommand = remotePathMapping(command);
 
+    // Share host mount points
+    mappedCommand.prepend(QLatin1String("export ") +
+                    QLatin1String("MER_SSH_HOST_SHARED_HOME=") +
```
Comment (vigejolla) [inline src/tools/merssh/command.cpp]: IIRC, we should no longer use QLatin1String

File: src/tools/merssh/cmakecommand.cpp [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+{
+    return QLatin1String("cmake");
+}
+
+int CMakeCommand::execute()
+{
+    if (arguments().contains(QLatin1String("--version"))
+            || arguments().contains(QLatin1String("--help"))
+            || arguments().contains(QLatin1String("-E capabilities"))) {
+        m_cacheFile = QLatin1String(Sfdk::Constants::CMAKE_QUERY_CACHE);
```
[the remark this answers is not in the document]
Comment (martyone) [reply src/tools/merssh/cmakecommand.cpp]: This is gone now

File: src/tools/merssh/command.cpp (hunk 3 above)
Comment (martyone) [reply src/tools/merssh/command.cpp]: This is gone now.

**Pull request merged**